source/appModules/notepad++.py

```diff
@@ -0,0 +1,27 @@
+#appModules/notepad++.py
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2017 NV Access Limited
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+
+"""App module for Notepad++
+"""
+
+import appModuleHandler
+from NVDAObjects import NVDAObject
+
+class NotepadPlusPlusScintillaEdit(NVDAObject):
+
+	def initOverlayClass(self):
+		pass
+
+	def shouldAcceptShowHideCaretEvent(self):
+		return False
+
+class AppModule(appModuleHandler.AppModule):
+
+	def event_NVDAObject_init(self, obj):
+		pass
+
+	def chooseNVDAObjectOverlayClasses(self, obj, clsList):
+		clsList.insert(0, NotepadPlusPlusScintillaEdit)

```

Comment:
This should only be inserted if the object is definitely the editor. Right now, this will be inserted for all objects; e.g. dialogs, etc. For example:

```
if obj.role == controlTypes.ROLE_EDITABLETEXT and obj.windowClassName == "Scintilla":
    clsList.insert(0, NotepadPlusPlusScintillaEdit)
```